Document/0x05i-Testing-Code-Quality-and-Build-Settings.md

```diff
@@ -553,3 +553,101 @@ class a$b
 - Xposed - http://repo.xposed.info/
 - Drozer - https://labs.mwrinfosecurity.com/assets/BlogFiles/mwri-drozer-user-guide-2015-03-23.pdf
 - GNU nm - https://ftp.gnu.org/old-gnu/Manuals/binutils-2.12/html_node/binutils_4.html
+
+
+### Checking for weaknesses in third party libraries/depdendencies
+
+#### Overview
+Android applications often make use of third party libraries. These third party libraries accelerate development as the developer has to write less code in order to solve a problem. There are two categories of libraries:
+- Libraries that are not (or should not) be packed within the actual production application, such as `Mockito` used for testing and libraries like `JavaAssist` used to compile certain other libraries.
+- Libraries that are packed within the actual production application, such as `Okhttp3`.
+
+These libraries can have the following two classes of unwanted side-effects:
+- A library can contain a vulnerability, which will make the application vulnerable. A good example is `OKhttp 2.6.4` in which TLS chain polution was possible to bypass pinning.

```

Comment:
to make it more clear, say "SSL pinning"

Reply:
fixed